lfs/client.go

```diff
@@ -374,7 +375,15 @@ func doApiRequestWithRedirects(req *http.Request, creds Creds, via []*http.Reque
 	}
 
 	if res.StatusCode == 307 {
-		redirectedReq, redirectedCreds, err := newClientRequest(req.Method, res.Header.Get("Location"))
+		redirectTo := res.Header.Get("Location")
+		locurl, err := url.Parse(redirectTo)
+		if err == nil && !locurl.IsAbs() {
+			// relative Location headers are allowed & should be supported
```

Comment:
I think this comment is unnecessary. It seems pretty clear that it's resolving a relative URL.